app/assets/javascripts/discourse/tests/integration/components/chart-test.js

```diff
@@ -0,0 +1,97 @@
+import { render } from "@ember/test-helpers";
+import { hbs } from "ember-cli-htmlbars";
+import { module, test } from "qunit";
+import { setupRenderingTest } from "discourse/tests/helpers/component-test";
+
+module("Integration | Component | Chart", function (hooks) {
+  setupRenderingTest(hooks);
+
+  function generateData() {
+    let data = [];
+    for (let i = 0; i < 10; i++) {
+      data.push({ x: i, y: Math.random() * 10 });
+    }
+    return data;
+  }
+
+  // We do this because comparing hashes at a pixel level is not reliable,
+  // even when calling a hash on a canvas that was not changing at all
+  // I was getting different results. PNG is more solid ground, and you can
+  // compare it visually easier too.
+  async function hashCanvasRenderedPNG(canvas) {
+    const dataURL = canvas.toDataURL("image/png");
+    const encoder = new TextEncoder();
+    const buffer = encoder.encode(dataURL);
+
+    const hashBuffer = await crypto.subtle.digest("SHA-256", buffer);
+
+    return Array.from(new Uint8Array(hashBuffer))
+      .map((byte) => byte.toString(16).padStart(2, "0"))
+      .join("");
+  }
```

Comment:
Do you need to calculate hashes or could you compare the data urls directly?

```js
assert.notStrictEqual(canvas.toDataURL(), previousDataURL);
```

Reply:
Good question, will check

Reply:
Yes I was able to do this